doc/automated_configuration.rst

```diff
@@ -0,0 +1,132 @@
+
+.. code-block:: bash
+
+    rsmsummarize generate --interactive > example_rsmsummarize.json
+
+The following screencast shows the interactive session (click to play):
+
+.. raw:: html
+    
+   <script id="asciicast-313149" src="https://asciinema.org/a/313149.js" data-autoplay="false" async></script>
+
+And here is the generated configuration file for ``rsmsummrize``:
+
+.. literalinclude:: assets/example_rsmsummarize.json
+    :language: javascript
+
+.. important:: 
+
+    If you want to include subgroup information in the reports for ``rsmtool``, ``rsmeval``, and ``rsmcompare``, you should add ``--subgroups`` to the command. For example, when you run ``rsmeval generate --interactive --subgroups`` you would be prompted to enter the subgroup column names and the ``general_sections`` list will also include subgroup-based sections. Since the ``subgroups`` option can accept multiple inputs, it is handled in the same way as the ``experiment_dirs`` option for ``rsmsummarize`` above. 
+
+We end with a list of important things to note about interactive generation:
+
+- Carefully read the instructions and notes displayed at the top when you first enter interactive mode.
+
+- If you do not redirect the output of the command to a file, the generated configuration file will simply be printed out.
+
+- You may see messages like "invalid option", "invalid file" on the bottom left while you are entering the value for a field. This is an artifact of real-time validation. For example, when choosing a training file for ``rsmtool``, the message "invalid file" may be displayed while you navigate to the actual file. Once you get to a valid file, this message should disappear.
+
+- Required fields will *not* accept a blank input (just pressing enter) and show an error message in the bottom left until a valid input is provided.
+
+- Optional fields will accept blank inputs since they have default values that will be used if no user input is provided. In some cases, default values are shown underlined in parentheses. 
+
+- The configuration files generated interactively contain comments (as indicated by ``// ...``). While RSMTool handles JSON files with comments just fine, you may need to remove the comments manually if you wish to use these files outside of RSMTool.
+
+Non-interactive Generation
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+For more advanced or experienced users who want to quickly get started with a dummy configuration file
+that they feel comfortable editing manually, RSMTool also provides the capability to generate configuration files non-interactively. To do so, simply omit the ``--interactive`` switch in the commands above. For example, to generate a dummy configuration file for ``rsmtool``, the command to run would be:
+
+.. code-block:: bash
+
+    rsmtool generate > dummy_rsmtool.json
+
+When running this command, the following warning would be printed out to stderr:
+
+.. code-block:: bash
```

Comment:
It's not a huge deal, but things like `for` are being highlighted since this is a "bash" code block. It there any way to keep it blocked off this way but to _not_ do the highlighting?

Reply:
Yes, good idea. I will do this.